osu.Game/Online/Multiplayer/StatefulMultiplayerClient.cs

```diff
@@ -52,6 +52,7 @@ public abstract class StatefulMultiplayerClient : Component, IMultiplayerClient,
 
         /// <summary>
         /// Whether the <see cref="StatefulMultiplayerClient"/> is currently connected.
+        /// This is NOT thread safe and usage should be scheduled.
```

Comment:
Yeah I'm having mixed feelings about this (as was sort of said on discord already). If this is unsafe, then why should the events (`LoadRequested`, `MatchStarted`) not be? Dunno, multiplayer threading seems all over the place, and this is a complication/exception that was not there before.

My gut feeling is that I'd probably prefer all public bindables to be safe to consume externally (by which I mean, guarantee all their value change callbacks are executed on the update thread).

I'm not sure I see in which scenarios this might backfire by changing execution order (feel free to point out some, which are probably obvious), and the flipside - not having to worry about a bound bindable N levels down potentially passing a value from the wrong thread - is not negligible IMO. We haven't had a case like that but if it ever happens it's going to suck.

Reply:
The main scenario is one which to be very aware of, since it is a common cause of use-after-disposal, memory leaks, and other weirdness:

`MultiplayerClient` has a lifetime that extends beyond the lounge / room / player drawables. We regularly rely on scheduling to ensure that callbacks aren't executed after a drawable is disposed. If having thread safety on `MultiplayerClient` is relied upon, there is 95% chance an action will be performed on a drawable post-disposal.

So you end up having to schedule twice - a second time against the target drawable to fix the above. Which adds up to two frames of delay, and makes the point of the first schedule (for `MultiplayerClient` to run everything external on the update thread) a bit of a moot point.

Reply:
Is that still a worry when it comes to bindables, though? If things are done properly (as in doing bound copy instead of direct subscription), any disposed drawable will unbind before that in `UnbindAllBindables`, or am I forgetting something else again?

Reply:
`UnbindAllBindables` can't be assumed to always be on the update thread. Yes normally that is the case, but there can be times when it isn't - when objects get left to the finalizer, as was the case in `BeatmapCarousel` previously.

It's unlikely for that to be an issue here, I suppose, but on the other hand in the context of `APIAccess` working the same way and even providing a similar xmldoc, I think it's reasonable to go with this for now...